### PR TITLE
Add nats

### PR DESCRIPTION
**Pull request by dreamer-nitj**
Add the nats docker bitnami image to the community images.
1) This diff only adds the basic scaffolding and doesn't yet
have the functional tests.

Test Plan:
[x] Ran the run.sh on ephemeral VM and it passed.

### Applicable issues

  - fixes #

### Checklist

- [ ] Added to the `README.md` using [readme-generator](https://github.com/rapidfort/community-images/readme-generator)

### CODE REVIEW

File: community_images/nats/bitnami/functional_test.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+# shellcheck disable=SC1091
+. "${SCRIPTPATH}"/../../common/helpers.sh
+
+HELM_RELEASE=rf-nats
+NAMESPACE=$(get_namespace_string "${HELM_RELEASE}")
+REPOSITORY=nats
+IMAGE_REPOSITORY=ankitrapidfort/"$REPOSITORY"
```
Comment (codervinod) [inline community_images/nats/bitnami/functional_test.sh:14]: change to RAPIDFORT_ACCOUNT

File: community_images/nats/bitnami/image.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: community-image
```
Comment (codervinod) [inline community_images/nats/bitnami/image.yml:1]: update this file and run ./scripts/image_generator.sh

File: community_images/nats/bitnami/run.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    report_pulls "${IMAGE_REPOSITORY}"
+
+    # waiting for pod to be ready
+    echo "waiting for pod to be ready"
+    kubectl wait deployments "${HELM_RELEASE}" -n "${NAMESPACE}" --for=condition=Available=True --timeout=10m
+
+    # get pod name
+    POD_NAME=$(kubectl -n "${NAMESPACE}" get pods -l app.kubernetes.io/name="$REPOSITORY" -o jsonpath="{.items[0].metadata.name}")
+
+    #... testing logic goes here....
```
Comment (codervinod) [inline community_images/nats/bitnami/run.sh:46]: remove this line

File: community_images/template/provider/run.sh [hunk 4]
```diff
@@ -20,13 +20,14 @@ else
...
 fi
 
+# the test function for launching container and basic testing
 test()
 {
     local IMAGE_REPOSITORY=$1
     local TAG=$2
     local NAMESPACE=$3
     local HELM_RELEASE="$REPOSITORY"-release
-
```
Comment (codervinod) [inline community_images/template/provider/run.sh:29]: remove this

File: community_images/nats/bitnami/functional_test.sh [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    kubectl wait pods "${HELM_RELEASE}"-0 -n "${NAMESPACE}" --for=condition=ready --timeout=10m
+
+    NATS_USER=$(kubectl get secret --namespace ${NAMESPACE} ${HELM_RELEASE} -o jsonpath='{.data.*}' | base64 -d | grep -m 1 user | awk '{print $2}' | tr -d '"')
+    NATS_PASS=$(kubectl get secret --namespace "${NAMESPACE}" "${HELM_RELEASE}" -o jsonpath='{.data.*}' | base64 -d | grep -m 1 password | awk '{print $2}' | tr -d '"')
+    echo -e "Client credentials:\n\tUser: $NATS_USER\n\tPassword: $NATS_PASS"
+
+    kubectl run nats-release-client --restart='Never' --env="NATS_USER=$NATS_USER" --env="NATS_PASS=$NATS_PASS" --image docker.io/bitnami/golang --namespace "${NAMESPACE}" --command -- sleep infinity
+    kubectl wait pods nats-release-client -n "${NAMESPACE}" --for=condition=ready --timeout=10m
+    echo "#!/bin/bash
+    GO111MODULE=off go get github.com/nats-io/nats.go
```
Comment (codervinod) [inline community_images/nats/bitnami/functional_test.sh:35]: is this same as commands.sh? Do we need to remove this section? Also can we make a function?

**Review by codervinod — COMMENTED** [review]
This looks good. one minor change requested. Also the generated files are not there, could you run the script?

File: community_images/nats/bitnami/functional_test.sh (hunk 5 above)
Comment (dreamer-nitj) [reply community_images/nats/bitnami/functional_test.sh:35]: yes, there are few dynamic values going into the commands.sh file, i have added a line to delete commands.sh once it's copied to container so commands.sh file is not going to be checked into the code

**Comment by dreamer-nitj** [comment]
> This looks good. one minor change requested. Also the generated files are not there, could you run the script?

done

File: community_images/nats/bitnami/image.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+name: nats
+official_name: NATS
+official_website: https://nats.io
+source_image_dh_link: bitnami/nats
```
Comment (codervinod) [inline community_images/nats/bitnami/image.yml:4]: source_image_provider: Bitnami

File: community_images/nats/bitnami/run.sh [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    # kill docker-compose setup container
+    docker-compose -f "${SCRIPTPATH}"/docker-compose.yml -p "${NAMESPACE}" down
+
+    # clean up docker file
+    rm -rf "${SCRIPTPATH}"/docker-compose.yml
+}
+
+declare -a BASE_TAG_ARRAY=("2.8.4-debian-11-r")
+
+build_images "${INPUT_REGISTRY}" "${INPUT_ACCOUNT}" "${REPOSITORY}" test "${PUBLISH_IMAGE}" "${BASE_TAG_ARRAY[@]}"
```
Comment (codervinod) [inline community_images/nats/bitnami/run.sh:79]: build_images "${INPUT_REGISTRY}" "${INPUT_ACCOUNT}" "${REPOSITORY}" "${REPOSITORY}" test "${PUBLISH_IMAGE}" "${BASE_TAG_ARRAY[@]}"

**Pull request merged**